src/tests/end2end/BlendStateTests.cpp

```diff
@@ -0,0 +1,1055 @@
+        .GetResult();
+
+    testPipeline = device.CreateRenderPipelineBuilder()
+        .SetSubpass(renderpass, 0)
+        .SetLayout(pipelineLayout)
+        .SetStage(nxt::ShaderStage::Vertex, vsModule, "main")
+        .SetStage(nxt::ShaderStage::Fragment, fsModule, "main")
+        .SetColorAttachmentBlendState(0, blendStates[0])
+        .SetColorAttachmentBlendState(1, blendStates[1])
+        // Blend state not set on third color attachment. It should be default
+        .SetColorAttachmentBlendState(3, blendStates[2])
+        .GetResult();
+
+
+    for (unsigned int c = 0; c < kColors.size(); ++c) {
+        RGBA8 base = kColors[((c + 31) * 29) % kColors.size()];
+        RGBA8 color0 = kColors[((c + 19) * 13) % kColors.size()];
+        RGBA8 color1 = kColors[((c + 11) * 43) % kColors.size()];
+        RGBA8 color2 = kColors[((c + 7) * 3) % kColors.size()];
+        RGBA8 color3 = kColors[((c + 13) * 71) % kColors.size()];
+
+        RGBA8 expected0 = color0 + base;
+        RGBA8 expected1 = color1 - base;
+        RGBA8 expected2 = color2;
+        RGBA8 expected3 = min(color3, base);
+
+        renderTargets[2].TransitionUsage(nxt::TextureUsageBit::OutputAttachment);
+        renderTargets[4].TransitionUsage(nxt::TextureUsageBit::OutputAttachment);
+        renderTargets[3].TransitionUsage(nxt::TextureUsageBit::OutputAttachment);
+        renderTargets[0].TransitionUsage(nxt::TextureUsageBit::OutputAttachment);
+
+        nxt::CommandBuffer commands = device.CreateCommandBufferBuilder()
+            .BeginRenderPass(renderpass, framebuffer)
+            .BeginRenderSubpass()
+            .SetRenderPipeline(basePipeline)
+            .SetBindGroup(0, MakeBindGroupForColors(std::array<RGBA8, 4>({ { base, base, base, base } })))
+            .DrawArrays(3, 1, 0, 0)
+
+            .SetRenderPipeline(testPipeline)
+            .SetBindGroup(0, MakeBindGroupForColors(std::array<RGBA8, 4>({ { color0, color1, color2, color3 } })))
+            .DrawArrays(3, 1, 0, 0)
+            .EndRenderSubpass()
+            .EndRenderPass()
+            .GetResult();
+
+        queue.Submit(1, &commands);
+
+        EXPECT_PIXEL_RGBA8_EQ(expected0, renderTargets[2], kRTSize / 2, kRTSize / 2) << "Attachment slot 0 using render target 2 should have been " << color0 << " + " << base << " = " << expected0;
+        EXPECT_PIXEL_RGBA8_EQ(expected1, renderTargets[4], kRTSize / 2, kRTSize / 2) << "Attachment slot 1 using render target 4 should have been " << color1 << " - " << base << " = " << expected1;
+        EXPECT_PIXEL_RGBA8_EQ(expected2, renderTargets[3], kRTSize / 2, kRTSize / 2) << "Attachment slot 2 using render target 3 should have been " << color2 << " = " << expected2 << "(no blending)";
+        EXPECT_PIXEL_RGBA8_EQ(expected3, renderTargets[0], kRTSize / 2, kRTSize / 2) << "Attachment slot 3 using render target 0 should have been min(" << color3 << ", " << base << ") = " << expected3;
+    }
+}
+
```

Comment:
Done. Default in D3D12 is 1,1,1,1 and in Metal it's 0,0,0,0. But I've made the default 0,0,0,0 so that by default we do less work. I believe the default is also 0 in Vulkan.